src/dfcx_scrapi/tools/google_drive_connector.py

```diff
@@ -0,0 +1,196 @@
+        data_pull = sheet.get_all_values()
+        data = pd.DataFrame(columns=data_pull[0], data=data_pull[1:])
+        return data
+
+    def dataframe_to_existing_sheet(
+        self, sheet_name: str, worksheet_name: str, dataframe: pd.DataFrame
+    ):
+        """Move Intent/TP data from a DataFrame to Google Sheets.
+
+        Args:
+          sheet_name: name of the sheets object to push data to.
+          worksheet_name: name of the worksheet in the sheets objec to push data to.
+          dataframe: pandas dataframe to push to the sheet and worksheet.
+
+        """
+        g_sheets = self.client.open(sheet_name)
+        worksheet = g_sheets.worksheet(worksheet_name)
+        set_with_dataframe(worksheet, dataframe)
+
+    def list_permissions(self, sheet_name: str):
+        """List existing permissions on a sheet
+
+        Args:
+          sheet_name: name of the sheets object to pull existing permissions from.
+
+        Returns:
+          permissions: list of permissions on the sheet object.
+        """
+        g_sheets = self.client.open(sheet_name)
+        permissions = g_sheets.list_permissions()
+        return permissions
+
+    def share_sheet(
+        self,
+        sheet_name: str,
+        email: str,
+        role: str = "writer",
+        perm_type: str = "user",
+        notify: bool = True,
+    ):
+        """Share an existing google sheet with emails.
+
+        Args:
+          sheet_name: name of the sheets object share
+          email: email to share the sheet with.
+          role: can be writer, reader or owner
+          perm_type: type of permissoin giving
+              user
+              group
+              domain
+              anyone
+          notify: true to notify email that the sheets object has been shared.
+        """
+        g_sheets = self.client.open(sheet_name)
+        g_sheets.share(email, role=role, perm_type=perm_type, notify=notify)
+
+    def add_worksheet(
+        self,
+        sheet_name: str,
+        worksheet_name: str,
+        rows: int = 100,
+        cols: int = 26,
+    ):
+        """Add worksheet to an existing google sheet.
+
+        Args:
+          sheet_name: name of the sheets object
+          worksheet_name: name of worksheet in the sheets object to add
+          rows: rows to add to the sheet
+          column: columns to add to the sheet
+        """
+        g_sheets = self.client.open(sheet_name)
+        g_sheets.add_worksheet(title=worksheet_name, rows=rows, cols=cols)
+
+    def delete_worksheet(self, sheet_name: str, worksheet_name: str):
+        """Delete worksheet from an existing google sheet.
+
+        Args:
+          sheet_name: name of the sheets object
+          worksheet_name: name of worksheet in the sheets object to delete
+        """
+        g_sheets = self.client.open(sheet_name)
+        worksheet = g_sheets.worksheet(title=worksheet_name)
+        g_sheets.del_worksheet(worksheet=worksheet)
+
+    def dataframe_to_new_sheet(
```

Comment:
@Greenford 

Can you group all of the similar methods together in this Class for better readability?

- create_x
- delete_x
- share_x
- list_x
- create_z
- delete_z
- Dataframe to X
- Dataframe to Y
- X to Dataframe

Also add an extra blank line at the end of the file.

Reply:
Done